mslib/msui/mpl_qtwidget.py

```diff
@@ -119,40 +127,34 @@ def __init__(self, fig=None, ax=None):
         # Sets the default fontsize parameters' values for topview from MSSDefaultConfig.
         self.topview_size_settings = config_loader(dataset="topview")
         # logging.debug("applying map appearance settings %s." % settings)
-        self.settings = {"draw_graticule": True,
-                         "draw_coastlines": True,
-                         "fill_waterbodies": True,
-                         "fill_continents": True,
-                         "draw_flighttrack": True,
-                         "draw_marker": True,
-                         "label_flighttrack": True,
-                         "tov_plot_title_size": "default",
-                         "tov_axes_label_size": "default",
-                         "colour_water": ((153 / 255.), (255 / 255.), (255 / 255.), (255 / 255.)),
-                         "colour_land": ((204 / 255.), (153 / 255.), (102 / 255.), (255 / 255.)),
-                         "colour_ft_vertices": (0, 0, 1, 1),
-                         "colour_ft_waypoints": (1, 0, 0, 1)}
-        self.appearance_settings = self.settings
+        self.set_settings({"draw_graticule": True,
+                           "draw_coastlines": True,
+                           "fill_waterbodies": True,
+                           "fill_continents": True,
+                           "draw_flighttrack": True,
+                           "draw_marker": True,
+                           "label_flighttrack": True,
+                           "tov_plot_title_size": "default",
+                           "tov_axes_label_size": "default",
+                           "colour_water": ((153 / 255.), (255 / 255.), (255 / 255.), (255 / 255.)),
+                           "colour_land": ((204 / 255.), (153 / 255.), (102 / 255.), (255 / 255.)),
+                           "colour_ft_vertices": (0, 0, 1, 1),
+                           "colour_ft_waypoints": (1, 0, 0, 1)})
         self.ax.figure.canvas.draw()
 
-    def get_map_appearance(self):
-        """
-        """
-        return self.appearance_settings
-
     def init_map(self, **kwargs):
-        self.map = mpl_map.MapCanvas(appearance=self.get_map_appearance(),
+        self.map = mpl_map.MapCanvas(appearance=self.get_settings(),
                                      resolution="l", area_thresh=1000., ax=self.ax,
                                      **kwargs)
 
         # Sets the selected fontsize only if draw_graticule box from topview options is checked in.
-        if self.appearance_settings["draw_graticule"]:
+        if self.settings["draw_graticule"]:
             try:
                 self.map._draw_auto_graticule(self.tov_als)
             except Exception as ex:
```

Comment:
we should add the specific exceptions here

Reply:
can probably be removed. this code is riddled with work-arounds for matplotlib 0.x

Reply:
Maybe make an issue for this? Investigate matplotlib-workarounds?

Reply:
https://github.com/Open-MSS/MSS/issues/1618